File: examples/php/establishments.php
<?php

declare(strict_types=1);

use Twila\ParceleMais\Config\ClientOptions;
use Twila\ParceleMais\Config\Environment;
use Twila\ParceleMais\Errors\ParceleMaisApiException;
use Twila\ParceleMais\ParceleMaisClient;
use Twila\ParceleMais\Establishments\BankAccountType;
use Twila\ParceleMais\Establishments\CreateEstablishmentRequest;
use Twila\ParceleMais\Establishments\DisbursementModel;
use Twila\ParceleMais\Establishments\EstablishmentAddress;
use Twila\ParceleMais\Establishments\EstablishmentBankAccount;
use Twila\ParceleMais\Establishments\EstablishmentOwner;
use Twila\ParceleMais\Establishments\ListEstablishmentsRequest;
use Twila\ParceleMais\Establishments\UpdateEstablishmentRequest;

$client = new ParceleMaisClient(new ClientOptions(
    getenv('PARCELEMAIS_CLIENT_ID'),
    getenv('PARCELEMAIS_CLIENT_SECRET'),
    Environment::STAGING
));

// Cadastrar loja
$result = $client->establishments->create(new CreateEstablishmentRequest(
    '12345678000199',
    'Loja Centro LTDA',
    'Loja Centro',
    DisbursementModel::ESTABLISHMENT_CHAIN,
    new EstablishmentOwner('Maria Souza', 'maria@loja.com.br', '+5511999998888'),
    new EstablishmentBankAccount('341', '1234', '56789', '0', BankAccountType::CURRENT),
    new EstablishmentAddress('Rua Exemplo', '100', 'Centro', 'São Paulo', 'SP', '01310100')
));

// Guarde o establishmentId — é ele que permite consultar, editar ou mudar a situação da loja depois.
$establishmentId = $result->establishmentId;
echo "Loja criada: {$establishmentId}\n";

// Buscar a loja
$establishment = $client->establishments->get($establishmentId);
echo "Loja {$establishment->tradeName}: " . ($establishment->isActive ? 'ativa' : 'inativa') . "\n";

// Listar as lojas ativas com "Centro" no nome fantasia
$active = $client->establishments->list(new ListEstablishmentsRequest('Centro', true));
echo count($active) . " loja(s) ativa(s)\n";

// Editar loja (o nome fantasia é sempre obrigatório; a razão social não muda)
$client->establishments->update($establishmentId, new UpdateEstablishmentRequest('Loja Centro Matriz'));

// Passar o desembolso para a conta da própria loja — a conta precisa existir antes
$client->establishments->updateBankAccount(
    $establishmentId,
    new EstablishmentBankAccount('341', '1234', '56789', '0', BankAccountType::CURRENT)
);

$client->establishments->update($establishmentId, new UpdateEstablishmentRequest(
    'Loja Centro Matriz',
    DisbursementModel::ESTABLISHMENT
));

// Inativar (a loja para de aceitar novos pedidos)
try {
    $client->establishments->deactivate($establishmentId);
} catch (ParceleMaisApiException $e) {
    if ($e->getStatusCode() === 404) {
        echo "Loja {$establishmentId} não pertence a este parceiro.\n";
    } else {
        throw $e;
    }
}

// Reativar
$client->establishments->activate($establishmentId);

File: examples/dotnet/establishments.cs
using System.Net;
using ParceleMais;
using ParceleMais.Errors;
using ParceleMais.Establishments.Models;

namespace ParceleMaisExamples;

public static class EstablishmentsExample
{
    public static async Task<Guid> CreateEstablishmentAsync(IParceleMaisClient client, CancellationToken ct = default)
    {
        var result = await client.Establishments.CreateAsync(new CreateEstablishmentRequest(
            Document: "12345678000199",
            LegalName: "Loja Centro LTDA",
            TradeName: "Loja Centro",
            DisbursementModel: DisbursementModel.EstablishmentChain,
            Owner: new EstablishmentOwner("Maria Souza", "maria@loja.com.br", "+5511999998888"),
            BankAccount: new EstablishmentBankAccount("341", "1234", "56789", "0", BankAccountType.Current),
            Address: new EstablishmentAddress("Rua Exemplo", "100", "Centro", "São Paulo", "SP", "01310100")), ct);

        // Persist EstablishmentId — it's required to read, edit or change the status of the establishment later.
        return result.EstablishmentId;
    }

    public static async Task<Establishment?> GetEstablishmentAsync(IParceleMaisClient client, Guid establishmentId, CancellationToken ct = default)
    {
        try
        {
            return await client.Establishments.GetAsync(establishmentId, ct);
        }
        catch (ParceleMaisApiException ex) when (ex.StatusCode == HttpStatusCode.NotFound)
        {
            Console.WriteLine($"Loja {establishmentId} não pertence a este parceiro.");
            return null;
        }
    }

    public static Task<IReadOnlyList<Establishment>> ListActiveEstablishmentsAsync(IParceleMaisClient client, string? tradeName = null, CancellationToken ct = default)
        => client.Establishments.ListAsync(new ListEstablishmentsRequest(TradeName: tradeName, IsActive: true), ct);

    public static Task RenameEstablishmentAsync(IParceleMaisClient client, Guid establishmentId, CancellationToken ct = default)
        => client.Establishments.UpdateAsync(establishmentId, new UpdateEstablishmentRequest("Loja Centro Matriz"), ct);

    public static async Task MoveDisbursementToEstablishmentAccountAsync(IParceleMaisClient client, Guid establishmentId, CancellationToken ct = default)
    {
        // The establishment needs its own bank account before it can receive the disbursement.
        await client.Establishments.UpdateBankAccountAsync(establishmentId,
            new EstablishmentBankAccount("341", "1234", "56789", "0", BankAccountType.Current), ct);

        await client.Establishments.UpdateAsync(establishmentId, new UpdateEstablishmentRequest(
            TradeName: "Loja Centro Matriz",
            DisbursementModel: DisbursementModel.Establishment), ct);
    }

    public static async Task CloseEstablishmentAsync(IParceleMaisClient client, Guid establishmentId, CancellationToken ct = default)
    {
        try
        {
            await client.Establishments.DeactivateAsync(establishmentId, ct);
        }
        catch (ParceleMaisApiException ex) when (ex.StatusCode == HttpStatusCode.NotFound)
        {
            Console.WriteLine($"Loja {establishmentId} não pertence a este parceiro.");
        }
    }

    public static Task ReopenEstablishmentAsync(IParceleMaisClient client, Guid establishmentId, CancellationToken ct = default)
        => client.Establishments.ActivateAsync(establishmentId, ct);
}
